staker/challenge-cache/cache_test.go

```diff
@@ -69,6 +74,148 @@ func TestCache(t *testing.T) {
 	}
 }
 
+func TestPrune(t *testing.T) {
+	ctx := context.Background()
+	basePath := t.TempDir()
+	if err := os.MkdirAll(basePath, os.ModePerm); err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		if err := os.RemoveAll(basePath); err != nil {
+			t.Fatal(err)
+		}
+	})
+	cache, err := New(basePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = cache.Init(ctx); err != nil {
+		t.Fatal(err)
+	}
+	key := &Key{
+		WavmModuleRoot: common.BytesToHash([]byte("foo")),
+		MessageHeight:  20,
+		StepHeights:    []uint64{0},
+	}
+	if _, err = cache.Get(key, 0); !errors.Is(err, ErrNotFoundInCache) {
+		t.Fatal(err)
+	}
+	t.Run("pruning non-existent dirs does nothing", func(t *testing.T) {
+		if err = cache.Prune(ctx, key.MessageHeight); err != nil {
+			t.Error(err)
```

Comment:
is it t.Error and not t.Fatal on urpose? (It's probably o.k. either way.. just making sure)

Reply:
Yes, so that it does not stop other runs